tests/integrations/file/test_File__download.py

```diff
@@ -22,12 +21,62 @@
 
 @pytest.fixture
 def mock_tools(mock_tools) -> ToolCache:
-    mock_tools.requests = mock.MagicMock(spec_set=requests)
+    mock_tools.httpx = mock.MagicMock(spec_set=httpx)
     # Restore move so the temporary file can be moved after downloaded
     mock_tools.shutil.move = mock.MagicMock(wraps=shutil.move)
     return mock_tools
 
 
+class _IteratorByteSteam(httpx.SyncByteStream):
+    """Shim that satisfies ``httpx.Response`` ``stream`` parameter type.
+
+    Cannot be replaced by any ``Iterable[bytes]`` because the base class requires
+    an explicit finalization method ``close``."""
+
+    def __init__(self, iterable: Iterable[bytes]) -> None:
+        self.iterable = iterable
+
+    def __iter__(self) -> Iterator[bytes]:
+        return iter(self.iterable)
+
+
+def _make_httpx_response(
+    *,
+    url: str,
+    status_code: int,
+    stream: list[bytes],
+    method: str = "GET",
+    headers: dict = {},
+) -> httpx.Response:
+    """Create a real ``httpx.Response`` with key methods wrapped by ``mock.Mock`` for spying.
+
+    Wrapped methods:
+        response.read
+        response.iter_bytes
+        response.headers.get
+    """
+    response = httpx.Response(
+        request=httpx.Request(
+            method=method,
+            url=httpx.URL(url),
+        ),
+        status_code=status_code,
+        headers=httpx.Headers(headers),
+        # Always use ``stream`` rather than content because it's more flexible
+        # even if the request is made non-streaming or the response is read with
+        # ``response.read()``, httpx will still consume the ``stream`` response
+        # content internally. This allows testing both the non-streaming and
+        # streaming download paths without needing to complicate the response params
+        stream=_IteratorByteSteam(stream),
```

Comment:
It took me a second to work out what was going on here - but once I did get it, it's a very nice simplification.